Single sign-on with OpenID Connect with Account Management Token (#117)
elasticpath/elasticpath-dev: guides/How-To/Authentication/get-single-sign-on-account-management-token.md

        }
    },
    // Other Fields...
}

```
### Get the list of configured single sign-on providers

To continue with the single sign-on flow, get the list of configured providers. From the list of configured providers presented to the account member, they choose which provider to authenticate with.

The following request uses the `data.relationships.authentication_realm.data.id` value from the previous step.

#### Request

```sh
curl -X GET https://useast.api.elasticpath.com/v2/authentication-realms/:realm-id/oidc-profiles/ \
     -H "Authorization: Bearer XXXX"
```

#### Response
```json
{
    // Other fields...
    "data": [
        {
            "links": {
                "authorization-endpoint": "https://epcc-integration.global.ssl.fastly.net/oidc-idp/login/stores/45b93fd2-3bfe-4ca2-ab66-924d054ff548/authentication-realms/447b6016-56f9-4fbc-8f13-ff12df1c98e9?elasticpath_commerce_cloud_profile_id=38da28ae-0391-4c0c-8e10-d737f1943371",
                // Other links...
            },
            "id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
            // Other fields...
        },
        // Other ODIC profiles...
    ],
}

```

> Make note of the `authorization-endpoint` link.

Reviewer: ```suggestion
Make a note of the `authorization-endpoint` link.
```

### Additional parameters to the authorization endpoint URL

The [OpenID Connect Specification](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) specifies some optional parameters that can be set to control the behaviour of the Identity Provider.

- `display` - Specifies how the authentication request is shown to the user.
- `prompt` - Controls whether the user is required to reauthenticate.
- `ui_locales` - Control the language that the login screen uses.

Commerce forwards these values to the configured Identity Provider, and you should consult your Identity Providers documentation to determine the allowed values.

### Log in to the OpenID Connect provider

The Composable Commerce application should now redirect the user to the modified `authorization-endpoint` link, which begins the single sign-on flow for the user.

### Assign the Account Member to an Account

> If for setting up for the first time the user will end up as an unassigned member. If the account member is already assigned to the account skip over to [Use the authorization code to retrieve an Account Management Authentication Token](#use-the-authorization-code-to-retrieve-an-account-management-authentication-token).

Reviewer: ```suggestion
When setting up for the first time, if the user is not yet assigned, they will be  categorized as an unassigned member. If the account member is already assigned, skip over to [Use the authorization code to retrieve an Account Management Authentication Token](#use-the-authorization-code-to-retrieve-an-account-management-authentication-token).
```

#### Request
```sh
curl -X GET https://useast.api.elasticpath.com/v2/accounts/:id/account-memberships/unassigned-account-members \
     -H "Authorization: Bearer XXXX"
```

#### Response
```json
{
    "data": [
        {
            "id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
            "type": "account_member",
            "name": "John Smith",
            "email": "jsmith@example.com",
            // Other fields...
        }
    ],
    // Other fields...
}
```

> Make a note of the `data[x].id` which will be used as the `account_member_id` when creating an account membership.

Reviewer: ```suggestion
Make a note of the `data[x].id`,  which will be used as the `account_member_id` when creating an account membership.
```

#### Request
```sh
curl -X POST https://useast.api.elasticpath.com/v2/accounts/:id/account-memberships \
     -H "Authorization: Bearer XXXX" \
     -H "Content-Type: application/json" \

>      -d $'{

Reviewer: ```suggestion
     -d $ {
```

Author: Not sure why we want to remove the quote here. We need these for the CURL call. other requests in the how to guides use single quotes here.

Reviewer: I thought it's not needed. It doesn't show in postman, does it?

Author: Postman does do it a little differently than what we have here. But anyways I tested without the single quote there and it breaks.

Reviewer: okay. Thank you for testing it in postman.  Please ignore my suggestion.

        "data":{
          "type": "account_membership",
          "account_member_id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
        }

>      }'

Reviewer: ```suggestion
     }
```

```
#### Response
```json
{
    "data": {
        "id": "YYYYYYYY-YYYYY-YYYY-YYYY-YYYYYYYYYY",
        "type": "account_membership",
        "relationships": {
            "account_member": {
                "data": {
                    "id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
                    "type": "account_member"
                }
            }
        }
        // Other fields...
    },
    // Other fields...
}

```

The account member will now be associated to an account.

### Use the authorization code to retrieve an Account Management Authentication Token

> After the [Log in to the OpenID Connect provider](#log-in-to-the-openid-connect-provider) step, the user follows a series of redirects to authenticate. Eventually they are redirected back to this Composable Commerce application, with some additional parameters as described in [Successful Authentication Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse).

Reviewer: ```suggestion
After the [Log in to the OpenID Connect provider](#log-in-to-the-openid-connect-provider) step, the user follows a series of redirects to authenticate. Eventually, they are redirected back to this Composable Commerce application, with some additional parameters as described in [Successful Authentication Response](https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse).
```

After validating that the `state` parameter matches the value sent in the [Add parameters to the Authorization Endpoint URL](#add-parameters-to-the-authorization-endpoint-url) section, the Composable Commerce application can use the `code` parameter in the URL, and the `code_verifier` from the [Generate Proof Key for Code Exchange (PKCE) Parameters](#generate-proof-key-for-code-exchange-pkce-parameters) section to get an [Account Management Authentication Token](https://elasticpath.dev/docs/api/accounts/post-v-2-account-members-tokens).

> Request an Account Management Authentication Token using the `oidc` as the `authentication_mechanism`. Use `code` generated as the `oauth_authorization_code`, `code_verifier` as `oauth_code_verifier`, and provide the redirect URI of the store which handled the response.

Reviewer: ```suggestion
Request an Account Management Authentication Token using the `oidc` as the `authentication_mechanism`. Use `code` generated as the `oauth_authorization_code`, `code_verifier` as `oauth_code_verifier`, and provide the redirect URI of the store, which handled the response.
```
